core/test/src/test_cpu.cpp

```diff
@@ -2480,4 +2482,94 @@ TEST_F(CpuTest, eor_absy_without_page_crossing) {
     EXPECT_EQ(expected, registers);
 }
 
+// ROL, ABS
+TEST_F(CpuAbsoluteTest, rol_abs_set_neg_clear_zero) {
+    registers.a = 0b00000000;
+    registers.p = Z_FLAG;
+    expected.a = 0b10101010;
+    expected.p = N_FLAG;
+    memory_content = 0b01010101;
+
+    run_read_instruction(ROL_ABS);
```

Comment:
rol absolute should take 6 cycles. Need to add a run_read_write_instruction member to the test fixture and add support for read write instructions there

Reply:
Ok, need to do my homework a bit more...